deploy-transact/lib/pantheon-node-signing.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+
+const EEAClient = require("web3-eea");
+const Web3 = require('web3');
+const axios = require('axios');
+const solc = require('solc');
+const join = require('path').join;
+const fs = require('fs-extra');
+const argv = require('yargs').argv;
+const verbose = argv.verbose;
+
+class PantheonNodeSigningHandler {
+  constructor(url, contractName) {
+    this.web3 = new EEAClient(new Web3(url), 2018);
+    this.url = url;
+    this.contractName = contractName;
+    this.abi;
+    this.bytecode;
+    this.account;
+  }
+
+ async getContract(){
+    this.account = await this.getAccount();
+    let tsSrc = fs.statSync(join(__dirname, `../contracts/${this.contractName}.sol`));
+    let tsBin;
+    try {
+        tsBin = fs.statSync(join(__dirname, `../contracts/${this.contractName}.bin`));
+    } catch(err) {
+        console.log("Compiled contract does not exist. Will be generated.");
+    }
+    let compiled;
+    if (!tsBin || tsSrc.mtimeMs > tsBin.mtimeMs) {
+        // source file has been modified since the last compile
+        let data = fs.readFileSync(join(__dirname, `../contracts/${this.contractName}.sol`));
+        compiled = solc.compile(data.toString(), 1);
+        fs.writeFileSync(join(__dirname, `../contracts/${this.contractName}.bin`), JSON.stringify(compiled));
+    } else {
+        compiled = JSON.parse(fs.readFileSync(join(__dirname, `../contracts/${this.contractName}.bin`)).toString());
+    }
+
+    let contract = compiled.contracts[`:${this.contractName}`];
+    this.abi = JSON.parse(contract.interface);
+    this.bytecode = '0x' + contract.bytecode;
+  }
+
+  async getAccount() {
+    console.log(`=> Connecting to target node: ${this.url}`);
+    let accounts = await this.web3.eth.getAccounts();
+    if (!accounts || accounts.length === 0) {
+      console.error("\tCan't find accounts in the target node");
+      process.exit(1);
+    }
+
+    console.log(`\tFound account in the target node: ${accounts[0]}`);
+    return accounts[0];
+  }
+
+  async deployContract(privateFor,privateFrom) {
+    await this.getContract();
+    let rpcInstance = axios.create({
```

Comment:
interesting, so `web3-eea` has `eea.getTransactionReceipt()` but does not support `eea.sendTransaction()`?

Reply:
The web3-eea supports only the following methods:

1. generatePrivacyGroup
2. getMarkerTransaction
3. getTransactionCount
4. getTransactionReceipt
5. sendRawTransaction

It doesn't support sendTransaction() using external signing using EthSigner